kDriveCore/Data/Api/DriveApiFetcher.swift

```diff
@@ -59,6 +59,10 @@ public struct PublicShareMetadata: Decodable {
 
     public let validUntil: TimeInterval?
     public let capabilities: Rights
+    
+    // TODO: Test parsing
+    public let isPasswordNeeded: Bool = false
+    public let isExpired: Bool = false
```

Comment:
Also not sure this will work. At which point do you want to test it ?

Reply:
Password protected will be in another PR, as well as expired.
I made sure to add completion steps to the feature PR.